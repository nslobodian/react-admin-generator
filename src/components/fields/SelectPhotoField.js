import React from 'react'
import Dropzone from 'react-dropzone'
import s from './FieldsStyles.scss'
import cn from 'classnames'

export default ({ label, input, meta, inline, showPreview = true, PreviewComponent, doubleProduct = false, previewValues }) => <div
  className={cn(s.selectPhotoWrap, inline && s.inlineStyles)}>
  {showPreview && (PreviewComponent
    ? <PreviewComponent preview={input.value.preview || input.value} {...previewValues} />
    : !!input.value && <img src={input.value.preview || input.value} alt='Preview' className='img-responsive m-b-1' />)}
  <Dropzone
    onDrop={acceptedFiles => {
      input.onChange(acceptedFiles[0])
    }}
    multiple={false}
    accept='image/*'
  >
    { input.value
      ? <button type='button' className='btn btn-default btn-block'>Change {label}</button>
      : <button type='button' className='btn btn-primary btn-block'>Upload {label}</button>
    }
    {inline && <div className='text-center text-muted'>{`Photo size 498x${doubleProduct ? '320' : '778'}px`}</div>}
  </Dropzone>
  {meta && meta.touched && meta.invalid &&
  <div className='alert alert-danger m-t-1' role='alert'>{meta.error}</div>}
</div>
