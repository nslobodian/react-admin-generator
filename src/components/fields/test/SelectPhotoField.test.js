import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SelectPhotoField from '../SelectPhotoField'

describe('(Components) SelectPhotoField', () => {
  it('Should render component properly', () => {
    const props = {
      label: 'label',
      input: {
        onChange: jest.fn(),
        value: {
          preview: 'preview',
        },
      },
      meta: {
        touched: jest.fn(),
        invalid: jest.fn(),
        error: 'error!',
      },
      inline: true,
      showPreview: false,
    }
    const wrapper = shallow(
      <SelectPhotoField
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  }),
  it('Should render component properly without PreviewComponent', () => {
    const props = {
      label: 'label',
      input: {
        onChange: jest.fn(),
        value: false,
      },
      meta: {
        touched: jest.fn(),
        invalid: jest.fn(),
        error: 'error!',
      },
      inline: true,
      PreviewComponent: jest.fn(),
    }
    const wrapper = shallow(
      <SelectPhotoField
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

