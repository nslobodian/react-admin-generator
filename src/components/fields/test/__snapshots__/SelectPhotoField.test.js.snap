// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Components) SelectPhotoField Should render component properly 1`] = `
<div
  className="selectPhotoWrap inlineStyles"
>
  <Dropzone
    accept="image/*"
    disableClick={false}
    disablePreview={false}
    maxSize={Infinity}
    minSize={0}
    multiple={false}
    onDrop={[Function]}
    preventDropOnDocument={true}
  >
    <button
      className="btn btn-default btn-block"
      type="button"
    >
      Change 
      label
    </button>
    <div
      className="text-center text-muted"
    >
      Photo size 500x325px
    </div>
  </Dropzone>
  <div
    className="alert alert-danger m-t-1"
    role="alert"
  >
    error!
  </div>
</div>
`;

exports[`(Components) SelectPhotoField Should render component properly without PreviewComponent 1`] = `
<div
  className="selectPhotoWrap inlineStyles"
>
  <mockConstructor
    preview={false}
  />
  <Dropzone
    accept="image/*"
    disableClick={false}
    disablePreview={false}
    maxSize={Infinity}
    minSize={0}
    multiple={false}
    onDrop={[Function]}
    preventDropOnDocument={true}
  >
    <button
      className="btn btn-primary btn-block"
      type="button"
    >
      Upload 
      label
    </button>
    <div
      className="text-center text-muted"
    >
      Photo size 500x325px
    </div>
  </Dropzone>
  <div
    className="alert alert-danger m-t-1"
    role="alert"
  >
    error!
  </div>
</div>
`;
